synapseclient/client.py

```diff
@@ -771,70 +776,64 @@ def setEndpoints(
+        Valid login combinations:
+            - authToken
+            - Profile-based authentication (from .synapseConfig)
 
-        1. .synapseConfig file (in user home folder unless configured otherwise)
-        2. User defined arguments during a CLI session
-        3. User's Personal Access Token (aka: Synapse Auth Token)
-            from the environment variable: SYNAPSE_AUTH_TOKEN
-        4. Retrieves user's authentication token from AWS SSM Parameter store (if configured)
+        If no login arguments are provided, login will attempt to authenticate from these sources:
+            1. .synapseConfig file (supports multiple profiles)
+            2. User-defined arguments during a CLI session
+            3. Environment variable (SYNAPSE_AUTH_TOKEN)
+            4. AWS Parameter Store (if configured)
 
         Arguments:
-            email:        Synapse user name (or an email address associated with a Synapse account)
-            authToken:    A bearer authorization token, e.g. a
-                [personal access token](https://python-docs.synapse.org/tutorials/authentication/).
-            silent:       Defaults to False.  Suppresses the "Welcome ...!" message.
+            email (str): Synapse user name or email.
+            authToken (str): Personal access token.
+            silent (bool): Suppress login message (default: False).
+            profile (str): Profile to use from .synapseConfig (default: "default").
 
-        Example: Logging in
-            Using an auth token:
+        **Note: You Do Not Need to Reinstantiate `Synapse` Every Time**
+        You only need to instantiate `Synapse` once per script/session before calling `.login()`.
+        If `Synapse` is already instantiated, just call `.login()` again to switch credentials.
 
-                syn.login(authToken="authtoken")
-                > Welcome, Me!
-
-            Using an auth token and username. The username is optional but verified
-            against the username in the auth token:
-
-                syn.login(email="my-username", authToken="authtoken")
-                > Welcome, Me!
+        Example:
+            - Logging in using a specific profile:
+            from synapseclient import Synapse
+            syn = Synapse() # Instantiate once
+            syn.login(profile="user1")
 
+        - Logging in with an authentication token:
+            from synapseclient import Synapse
+            syn = Synapse() # Instantiate once
+            syn.login(authToken="your_auth_token")
         """
-        # Note: the order of the logic below reflects the ordering in the docstring above.
-
-        # Check version before logging in
-        if not self.skip_checks:
-            version_check()
-
-        # Make sure to invalidate the existing session
+        # Ensure previous session is cleared
         self.logout()
 
+        user_login_args = UserLoginArgs(profile=profile or "default", username=email, auth_token=authToken)
```

Comment:
```suggestion
        user_login_args = UserLoginArgs(profile=profile, username=email, auth_token=authToken)
```

You are already setting a default for `profile` in the argument of the function: `profile: str = "default"`